### ISSUE TITLE
Notifications

### ISSUE THREAD

**Issue by buresdv** [buresdv/Mlem#45]
Many users are requesting an implementation of notifications for things such as:
- Replies to their posts and comments
- New posts in followed communities

## The Problem
I have no clue how to implement notifications. Perhaps sometime in the future.

**Comment by ShadowJonathan** [comment]
> I have no clue how to implement notifications.

Apple has a thing called a "Notification Extension" that is able to run in the background, and run the code to create notifications, however, to process and publish push notifications, these need to come through the APNS (Apple Push Notification Service), and only then Apple will kick off this extension code.

Apollo solves this by asking for a subscription ("Ultra") to basically proxy your user account via their servers, and to fetch notifications regularly, and use APNS to get Apollo to notify you.

**Comment by buresdv** [comment]
Thank you, this seems very complicated.

I think this issue is not a priority atm, so we can come back to it later.

**Comment by ShadowJonathan** [comment]
https://developer.apple.com/documentation/uikit/app_and_environment/scenes/preparing_your_ui_to_run_in_the_background/using_background_tasks_to_update_your_app

https://developer.apple.com/documentation/usernotifications/setting_up_a_remote_notification_server/sending_notification_requests_to_apns

**Comment by DaveAlton** [comment]
Agreed it’s complicated. Having a tab like Apollo’s “Inbox” tab would be incredibly helpful in The meantime. 

All the app would need to do is poll the notifications frequently from the server, then add a badge on the tab when there are new ones. Users wouldn’t get notifications unless they’re in the app, but at least they could know when someone has replied to them. This could be really helpful to drive engagement in Lemmy.
